qcodes/instrument_drivers/Lakeshore/lakeshore_base.py

```diff
@@ -0,0 +1,302 @@
+from typing import Dict, ClassVar
+import logging
+from collections import OrderedDict
+import time
+from bisect import bisect
+
+from qcodes import VisaInstrument, InstrumentChannel, ChannelList
+from qcodes.instrument.parameter import Parameter
+from qcodes.utils import validators
+
+log = logging.getLogger(__name__)
+
+class GroupParameter(Parameter):
+    def __init__(self, name, instrument, **kwargs):
+        self.group = None
+        super().__init__(name, instrument, **kwargs)
+
+    def set_raw(self, value, **kwargs):  # pylint: disable=E0202
+        self.group.set(self, value)
+
+    def get_raw(self, result=None):  # pylint: disable=E0202
+        if not result:
+            self.group.update()
+            return self.raw_value
+        else:
+            return result
+
+
+class Group():
+    def __init__(self, parameters, set_cmd=None, get_cmd=None,
+                 get_parser=None, separator=',', types=None) -> None:
+        self.parameters = OrderedDict((p.name, p) for p in parameters)
+        self.instrument = parameters[0].root_instrument
+        for p in parameters:
+            p.group = self
```

Comment:
There should be a public method called `set_group` on the `GroupParameter` class to do this. Do not assign attributes directly like this.

Reply:
In fact, if you make a parameter group like so: 

```
self._group_param = Group(
	set_cmd="bla {{a}} {{b}}"
)

self.add_parameter(
	"a", 
	group=self._group_param, 
	parameter_class=GroupParameter
)
```
Then line 19 will not be necessary and we will not need to set the `group` attribute because we pass it to the constructor. I think this is more clear.

Reply:
There is, as far as I can see, a mutual dependence of the parameter and the group: The parameter has to know the group, and the group has to know the parameter. 
This is because we want to call set/get on the parameter and it and needs the values of the other parameters to accomplish that.
So this double link is created in an init method. This can equally happen in the parameter or in the group. In your example the parameter is aware of the group but the group is not aware of the parameter.
I chose to put this linking into the group because it makes the code a little more compact and readable for my taste... But I can be convinced otherwise if you include the previous aspect in your argument.
concerning the attribute I wrote you a PM.

Reply:
I think you are right about this. I think if you clean this code up a little (e.g. add docstrings, proper type hinting) it can be approved and merged. But I am still unsure if we should have a setter for the group or not. I read the link you send me on PM, but I still think having a setter will make things more clear to fellow programmers.